doc/sphinx/practical-tools/coq-commands.rst

```diff
@@ -42,16 +42,37 @@ toplevel with the command ``Coqloop.loop();;``.
 Batch compilation (coqc)
 ------------------------
 
-The ``coqc`` command takes a name *file* as argument. Then it looks for a
-file named *file*.v, and tries to compile it into a
-*file*.vo file (See :ref:`compiled-files`).
+The ``coqc`` command compiles a Coq proof script file with a ".v" suffix
+to create a compiled file with a ".vo" suffix.  (See :ref:`compiled-files`.)
+The components of the filename must be valid Coq identifiers as described in
+:ref:`lexical-conventions`; they should contain only letters, digits or
+underscores (_) with a ".v" suffix on the final component.
+For example ``/bar/foo/toto.v`` is valid, but ``/bar/foo/to-to.v`` is not.
 
-.. caution::
+We recommend specifying a logical directory name (which is also the module name)
+with the `-R` or the `-Q` options.
+Generally we recommend using provided utilities such as `make` to build Coq projects.
+See :ref:`building_coq_project`.
 
-   The name *file* should be a regular Coq identifier as defined in Section :ref:`lexical-conventions`.
-   It should contain only letters, digits or underscores (_). For example ``/bar/foo/toto.v`` is valid,
-   but ``/bar/foo/to-to.v`` is not.
+.. example:: Compiling and loading a single file
 
+   To compile `foo.v` as part of a module `Mod1` that is rooted
+   at `.` (i.e. the directory containing `foo.v`), run `coqc -Q . Mod1 foo.v`.
+
+   To load the module within `coqtop` or `coqide`, use:
+
+      .. coqdoc::
+         Add LoadPath "PATH" as Mod1.
```

Comment:
Please don't use this command in example code, it is not recommended to setup your workspace with vernacular commands, but instead using the corresponding flags.

Reply:
Meaning in _CoqProject or on the command if you must?  At the moment, both require restarting coqtop/coqide.  Perhaps coqide could be modified so it doesn't need to be restarted.

I agree `Add LoadPath` is not great, but it doesn't require restarting CoqIDE, which is a bit disruptive.  So maybe reasonable for quick experiments.

Reply:
For coqide putting the flags in coqproject then closing/reopening the .v file should work.

Reply:
Less bad, thanks, but it should be more seamless in CoqIDE.  Opening a new scratch buffer doesn't reload the info, which I expected.

Reply:
Looks like CoqIDE lets you set the name of the project file in preferences.  `coqtop` doesn't seem to apply the default name, nor do I see a command line option for providing the name.  That's a bug, too, even if normal users won't use coqtop.

Reply:
>That's a bug, too, even if normal users won't use coqtop.

What does that mean? coqtop is kept for proffgeneral and that's about it.

Reply:
We use it in the development process and tests, too.

Reply:
`coqtop` and `coqc` only have `-R` / `-Q` options that you must pass manually. The idea is that users are never supposed to call these commands directly. A user should use an editor, which should load `_CoqProject` and `coq_makefile` or `dune` to build.